src/test/ui/attributes/doc-attr.stderr

```diff
@@ -13,6 +13,51 @@ LL | #![deny(warnings)]
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>
 
+error: unknown `doc` attribute
+  --> $DIR/doc-attr.rs:12:7
+   |
+LL | #[doc(123)]
+   |       ^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>
+
+error: unknown `doc` attribute
+  --> $DIR/doc-attr.rs:15:7
+   |
+LL | #[doc("hello", "bar")]
+   |       ^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>
+
+error: unknown `doc` attribute
+  --> $DIR/doc-attr.rs:15:16
+   |
+LL | #[doc("hello", "bar")]
+   |                ^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>
+
+error: unknown `doc` attribute ``
+  --> $DIR/doc-attr.rs:20:7
+   |
+LL | #[doc(foo::bar, crate::bar::baz = "bye")]
+   |       ^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>
+
+error: unknown `doc` attribute ``
```

Comment:
How do I display the `Path`? There doesn't seem to be a Display impl.

Note that this is not a regression; this test is just exposing the previous behavior.

Reply:
Instead of `name_or_empty`, could use `name_value_literal_span` and then get the snippet for that `Span`? Or is [`path_to_string`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast_pretty/pprust/fn.path_to_string.html) accessible here?

Reply:
I don't think `rustc_ast_pretty` is accessible here, though I could add it as a dependency. I'm trying out the `span_to_snippet` approach now.

Reply:
Both `span_to_snippet` and `path_to_string` worked. I'm leaning towards `path_to_string` because I think it will handle `foo :: bar:: baz` gracefully as `foo::bar::baz`, but which one would you prefer I use? You can see what the code looks like with `span_to_snippet` vs `path_to_string` here: 8f40e11

Reply:
I'd normally want to lean against introducing more dependencies between our crates, but depending on `rust_ast_pretty` when we already depend on `rust_ast` is probably fine. Let's stick with `path_to_string`.

Reply:
It looks like `rustc_passes` already depends transitively on `rustc_ast_pretty` anyway.